sdl_android/src/main/java/com/smartdevicelink/protocol/SdlProtocol.java

```diff
@@ -0,0 +1,1319 @@
+        if(SessionType.RPC.equals(serviceType)){
+            if(connectedPrimaryTransport != null) {
+                header.setTransportRecord(connectedPrimaryTransport);
+            }
+            //This is going to be our primary transport
+            header.putTag(ControlFrameTags.RPC.StartService.PROTOCOL_VERSION, MAX_PROTOCOL_VERSION.toString());
+            handlePacketToSend(header);
+            return; // We don't need to go any further
+        }else if(serviceType.equals(SessionType.NAV)){
+            if(iSdlProtocol != null){
+                VideoStreamingParameters videoStreamingParameters = iSdlProtocol.getDesiredVideoParams();
+                if(videoStreamingParameters != null) {
+                    ImageResolution desiredResolution = videoStreamingParameters.getResolution();
+                    VideoStreamingFormat desiredFormat = videoStreamingParameters.getFormat();
+                    if (desiredResolution != null) {
+                        header.putTag(ControlFrameTags.Video.StartService.WIDTH, desiredResolution.getResolutionWidth());
+                        header.putTag(ControlFrameTags.Video.StartService.HEIGHT, desiredResolution.getResolutionHeight());
+                    }
+                    if (desiredFormat != null) {
+                        header.putTag(ControlFrameTags.Video.StartService.VIDEO_CODEC, desiredFormat.getCodec().toString());
+                        header.putTag(ControlFrameTags.Video.StartService.VIDEO_PROTOCOL, desiredFormat.getProtocol().toString());
+                    }
+                }
+            }
+        }
+        if(transportPriorityForServiceMap == null
+                || transportPriorityForServiceMap.get(serviceType) == null
+                || transportPriorityForServiceMap.get(serviceType).isEmpty()){
+            //If there is no transport priority for this service it can be assumed it's primary
+            header.setTransportRecord(connectedPrimaryTransport);
+            handlePacketToSend(header);
+            return;
+        }
+        int transportPriority = transportPriorityForServiceMap.get(serviceType).get(0);
+        if(transportPriority == PRIMARY_TRANSPORT_ID){
+            // Primary is favored, and we're already connected...
+            header.setTransportRecord(connectedPrimaryTransport);
+            handlePacketToSend(header);
+        }else if(transportPriority == SECONDARY_TRANSPORT_ID) {
+            // Secondary is favored
+            for(TransportType secondaryTransportType : supportedSecondaryTransports) {
+
+                if(!requestedSecondaryTransports.contains(secondaryTransportType)){
+                    // Secondary transport is not accepted by the client
+                    continue;
+                }
+
+                if(activeTransports.get(serviceType).getType().equals(secondaryTransportType)){
```

Comment:
Might be a symptom of another issue, but a `NullPointerException` can happen here if `requiresHighBandwidth` is false.